### ISSUE TITLE
bug running pipeline on O2

### ISSUE THREAD

**Issue by rohanmaddamsetti** [debbiemarkslab/EVcouplings#138]
I get this error message:

Traceback (most recent call last):
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/scope.py", line 179, in resolve
    return self.resolvers[key]
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/collections/__init__.py", line 881, in __getitem__
    return self.__missing__(key)            # support subclasses that define __missing__
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/collections/__init__.py", line 873, in __missing__
    raise KeyError(key)
KeyError: 'probability'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/scope.py", line 190, in resolve
    return self.temps[key]
KeyError: 'probability'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/evcouplings/utils/pipeline.py", line 454, in execute_wrapped
    outcfg = execute(**config)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/evcouplings/utils/pipeline.py", line 174, in execute
    outcfg = runner(**incfg)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/evcouplings/compare/protocol.py", line 1022, in run
    return PROTOCOLS[kwargs["protocol"]](**kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/evcouplings/compare/protocol.py", line 630, in standard
    ec_table, d_intra, d_multimer, **kwargs
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/evcouplings/compare/protocol.py", line 207, in _make_contact_maps
    ec_set = ecs_longrange.query("probability >= @c")
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/frame.py", line 2116, in query
    res = self.eval(expr, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/frame.py", line 2186, in eval
    return _eval(expr, inplace=inplace, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/eval.py", line 262, in eval
    truediv=truediv)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 727, in __init__
    self.terms = self.parse()
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 744, in parse
    return self._visitor.visit(self.expr)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 313, in visit
    return visitor(node, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 319, in visit_Module
    return self.visit(expr, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 313, in visit
    return visitor(node, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 322, in visit_Expr
    return self.visit(node.value, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 313, in visit
    return visitor(node, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 644, in visit_Compare
    return self.visit(binop)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 313, in visit
    return visitor(node, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 416, in visit_BinOp
    op, op_class, left, right = self._maybe_transform_eq_ne(node)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 354, in _maybe_transform_eq_ne
    left = self.visit(node.left, side='left')
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 313, in visit
    return visitor(node, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/expr.py", line 430, in visit_Name
    return self.term_type(node.id, self.env, **kwargs)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/ops.py", line 58, in __init__
    self._value = self._resolve_name()
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/ops.py", line 75, in _resolve_name
    res = self.env.resolve(self.local_name, is_local=self.is_local)
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_o2/lib/python3.5/site-packages/pandas/core/computation/scope.py", line 192, in resolve
    raise compu.ops.UndefinedVariableError(key, is_local)
pandas.core.computation.ops.UndefinedVariableError: name 'probability' is not defined

**Comment by aggreen** [comment]
This issue was fixed by #130 and is occurring because we have not yet updated the evcouplings distributions on o2. Watch for an email from @sacdallago later today about those updates.

**Comment by sacdallago** [comment]
Please change the env in your config to `evcouplings_production`, or even `evcouplings_develop`. Thanks :)


If this persists, please reopen the issue.

**Comment by rohanmaddamsetti** [comment]
When I change the env to evcouplings_stable or evcouplings_develop I get the following error:

Traceback (most recent call last):
  File "/n/groups/marks/software/anaconda_o2/bin/evcouplings", line 11, in <module>
    load_entry_point('evcouplings==0.0.2', 'console_scripts', 'evcouplings')()
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 565, in load_entry_point
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2598, in load_entry_point
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2258, in load
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2264, in resolve
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/evcouplings/utils/app.py", line 21, in <module>
    import evcouplings.management.computeJob
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/evcouplings/management/__init__.py", line 1, in <module>
    from evcouplings.management.protocol import *
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/evcouplings/management/protocol.py", line 2, in <module>
    from evcouplings.management.dumper import get_dumper
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/evcouplings/management/dumper/__init__.py", line 1, in <module>
    from evcouplings.management.dumper.AzureDumper import AzureDumper
  File "/n/groups/marks/software/anaconda_o2/lib/python3.6/site-packages/evcouplings/management/dumper/AzureDumper.py", line 4, in <module>
    from azure.storage.blob import BlockBlobService, ContentSettings, PublicAccess
ModuleNotFoundError: No module named 'azure'

**Comment by sacdallago** [comment]
@rohanmaddamsetti sorry about this error, but this is extremely odd. It seems like you are using my branch of evcouplings, which is installed solely in my environments. I clearly installed evcouplings via pip in evcouplings_stable, so if that error happens, there is something fundamentally broken somewhere.

1. Where are you getting this error? If at the level of running evcouplings from the terminal, there clearly is something wrong and I need to look into this ASAP.

2. If it's at the level of your job failing, and you pasting here the content of the `xxx.failedx file, I suspect you haven't change the "source activate" instruction in your config file.

**Comment by b-schubert** [comment]
I fixed it.
@christian: you must have installed your branch instead of the pip version.

I fixed it for evcouplings_stable (but not for evcouplings_development)

**Comment by sacdallago** [comment]
@b-schubert I'm getting very suspicious, honestly. Look at the stack trace of @rohanmaddamsetti , evcouplings is a binary (fine) but it is not in an env (wrt. to the first stack trace). I'm pretty, pretty sure I ran `pip install evcouplings` on stable, there's a very slim chance I could have mistyped that :)
Additionally, I'm noticing that I get errors myself with my development environment which now seem to make sense. Specifically, what I was telling you this morning. I do the following:

1. `source activate evcouplings_backend` (my backend env)
1. I pull my dev version of evcouplings in `~/cev`
2. `pip install --no-deps --upgrade ~/cev`
3. Run some job

The result of that varies, but very often the job fails to be submitted entirely because some function that I implemented is "not implemented" by evcouplings. Which could mean that it's calling up a different version than what I installed.

I want to spend some time with you on Monday to test this out if that's ok with you?
